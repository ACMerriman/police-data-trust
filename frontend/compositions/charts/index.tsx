import { Circle, Square } from "./chartSymbols"
import { KeyEntryWrapper } from "./chartKey"
export { Circle, Square, KeyEntryWrapper}
