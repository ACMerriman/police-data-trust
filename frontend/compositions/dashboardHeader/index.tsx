import * as React from "react"
import NPDCLogo from '../../shared-components/logo'
import HamburgerMenu from './assets/Vector.svg'
import DesktopNav from './desktopNav'
import styles from './dashboardHeader.module.css'

export default function DashboardHeader() {
  const {
    wrapper,
    backgroundBanner,
    leftHeader,
    logoContainer,
    titleContainer,
    mobileTitle,
    desktopTitle,
    rightHeader,
    nav,
    buttonContainer,
    mobileNav,
    desktopNav,
  } = styles;

  return (
    <header className={wrapper}>
      <div className={backgroundBanner}>
        <div className={leftHeader}>
          <div className={logoContainer}>
            <NPDCLogo />
          </div>
          <div className={titleContainer}>
            <h2 className={mobileTitle}>N.P.D.C.</h2>
            <h2 className={desktopTitle}>National Police Data Coalition</h2>
            <p>The national index of police incidents</p>
          </div>
        </div>

        <div className={rightHeader}>
          <nav className={nav}>
            <div className={mobileNav}>
              <HamburgerMenu />
            </div>
            <div className={desktopNav}>
              <DesktopNav />
            </div>
          </nav>
          <div className={buttonContainer}>
            <button className="primaryButton" type="button">
              DONATE
            </button>
          </div>
        </div>
      </div>
    </header>
  )
}
