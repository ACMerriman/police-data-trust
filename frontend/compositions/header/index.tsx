import React, { useState } from "react"
import NPDCLogo from './assets/NPDCLogo.svg'
import styles  from './header.module.css'



export default function Header() {
  let [selected, setSelected] = useState('Search');

  const handleNavChange = (e: any) => {
    let value = e.target.innerText;
    setSelected(value);
  }

  const { wrapper, banner, logoTitle, titleContainer, nav, tab, donate, navDonate} = styles;

  return (
    <header className={wrapper}>
      <div className={banner}>
        <div className={logoTitle}>
          <NPDCLogo />
          <div className={titleContainer}>
            <h2>National Police Data Coalition</h2>
            <p>The national index of police incidents</p>
          </div>
        </div>
        <div className={navDonate}>
          <nav>
            <ul className={nav}>
              <li className={selected === 'Search' ? tab : ''} onClick={handleNavChange}>Search</li>
              <li className={selected === 'Profile' ? tab : ''} onClick={handleNavChange}>Profile</li>
              <li className={selected === 'About' ? tab : ''} onClick={handleNavChange}>About</li>
            </ul>
          </nav>
          <button className={donate}type="button">DONATE</button>
        </div>
      </div>
    </header>
  )
}
