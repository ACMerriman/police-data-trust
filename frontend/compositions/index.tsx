import DashboardHeader from "./dashboard-header/dashboard-header"
import EnrollmentCallToAction from './enrollment-call-to-action/enrollment-call-to-action'
import EnrollmentHeader from './enrollment-header/enrollment-header'
import PasswordAid from './password-aid/password-aid'

export { 
  DashboardHeader,
  EnrollmentCallToAction,
  EnrollmentHeader,
  PasswordAid
}
