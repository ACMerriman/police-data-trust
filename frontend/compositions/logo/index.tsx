import React from "react"
import LogoSVG from "./assets/NPDCLogo.svg"

export default function Logo() {
  return (
    <div className="Logo">
      <LogoSVG />
    </div>
  )
}
