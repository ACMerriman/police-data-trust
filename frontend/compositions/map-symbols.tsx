//@ts-nocheck
import * as d3 from "d3"
import { useEffect, useRef, useState } from "react"

export function MarkerLayer({ data, projection }) {
  const ref = useRef()
  const svg = d3.select(ref.current)

  const [circles, setCircles] = useState([])
  const path = d3.geoPath()
  const circleGenerator = d3.geoCircle()

  useEffect(() => {
    if (!data) return

    const populationScale = d3
      .scaleLinear()
      .domain(d3.extent(data.data.features, (d) => d.properties.population))
      .range([1, 50])

    const dataCircs = data.data.features.slice(0, 100).map((d, i) => {
      const circ = circleGenerator({
        r: populationScale(d.properties.population),
        center: d.geometry.coordinates,
      })
      return {
        center: projection(d.geometry.coordinates),
        path: path(circ),
        population: populationScale(d.properties.population),
      }
    })

    setCircles(dataCircs)
  }, [data])

  return (
    <svg ref={ref} viewBox="0, 0, 1200, 700" className="marker-layer">
      {circles.map((c, i) => (
        <circle
          className="circle-marker"
          cx={c.center[0]}
          cy={c.center[1]}
          r={c.population}
          key={`${i}${c.population}`}
          fill="transparent"
          stroke="green"
          strokeWdth="3"
        />
      ))}
    </svg>
  )
}
