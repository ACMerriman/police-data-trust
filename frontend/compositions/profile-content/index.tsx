import ProfileInfo from './profile-info'
import ProfileType from './profile-type'
import SavedSearches from './saved-searches'
import SavedResults from './saved-results'


export {
  ProfileInfo,
  ProfileType,
  SavedResults,
  SavedSearches
}