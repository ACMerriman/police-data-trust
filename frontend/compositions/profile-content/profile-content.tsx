import * as React from 'react'
import Link from 'next/link'
import styles from './profile-content.module.css'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faEdit } from "@fortawesome/free-solid-svg-icons"
import { UserType, profileTypeContent, emptyUser } from '../../models/profile'

interface UserProfileProps {
  userData: UserType
}


export function ProfileInfo({ userData = emptyUser }: UserProfileProps) {
  const [editMode, setEditMode] = React.useState(false)
  const { profileData, sectionTitle, row } = styles

  function handleSubmit(e: React.FormEvent) {
    console.log('edit profile form submitted')
  }

  if (editMode) {
    return (
      <div className={profileData}>
        <header className={sectionTitle}>Edit Your Account Information</header>
        <form onSubmit={handleSubmit} >
          <div className={row}>
            <label htmlFor="firstName">First name:</label>
            <input type="text" id="firstName" placeholder={userData.firstName} />
            <label htmlFor="lastName">Last name:</label>
            <input type="text" id="lastName" placeholder={userData.lastName} />
          </div>
          <div className={row}>
            <label htmlFor="email">Email address:</label>
            <input type="email" id="email" placeholder={userData.email} />
            <label htmlFor="phone">Phone number:</label>
            <input type="phone" id="phone" placeholder={userData.phone} />
          </div>
          <div className={row}>
            <label htmlFor="newPassword">Password:</label>
            <input type="password" id="newPassword" />
            <label htmlFor="confirmPassword">Confirm New Password</label>
            <input type="password" id="confirmPassword" />
          </div>
          <div className={row}>
            <button onClick={() => setEditMode(false)}>Cancel</button>
            <input type="submit" value="Save Changes" />
          </div>
         

        </form>
      </div>
    )
  } else {
    const { editButton, label, dataField } = styles
    return (
      <div className={profileData}>
        <header className={sectionTitle}>
          Your Account Information
          <FontAwesomeIcon 
            icon={faEdit} 
            size="1x" 
            className={editButton}
            onClick={() => setEditMode(true)} />
        </header>
        <div className={row}>
          <span className={label}>First name:</span>
          <div className={dataField}>{userData.firstName}</div>
          <span className={label}>Last name:</span>
          <div className={dataField}>{userData.lastName}</div>
        </div>
        <div className={row}>
          <span className={label}>Email address:</span>
          <div className={dataField}>{userData.email}</div>
          <span className={label}>Phone number:</span>
          <div className={dataField}>{userData.phone}</div>
        </div>
        <div className={row}>
          <span className={label}>Password:</span>
          <div className={dataField}>********</div>
        </div>
        
      </div>
    )
  }
  
}

export const ProfileType = ({ userData = emptyUser }: UserProfileProps) => {
  const {title, content, linkText, linkPath } = profileTypeContent[userData.role]
  const { profileData, sectionTitle, profileTypeText, profileTypeLink, description } = styles
  return (
    <div className={profileData}>
      <header className={sectionTitle}>Your Profile Type</header>
      <div className={profileTypeText}>{title}</div>
      <p className={description}>{content}</p>
      <div className={profileTypeLink}>
        <Link href={linkPath}>{linkText}</Link>
      </div>
    </div>
  )
}


// placeholders
export const SavedSearch = ({ userData = emptyUser }: UserProfileProps) => {

  return (
    <div className={styles.profileData}>
      <header className={styles.sectionTitle}>Saved Searches</header>
    </div>
    
    )
}

export const SavedResults = ({ userData = emptyUser }: UserProfileProps) => {
  return (
    <div className={styles.profileData}>
      <header className={styles.sectionTitle}>Saved Results</header>
    </div>
    
    )
}
