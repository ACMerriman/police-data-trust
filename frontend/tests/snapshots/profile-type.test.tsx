import * as React from "react"
import { ProfileType } from "../../compositions/profile-content"
import renderer from "react-test-renderer"

it("renders Profile: Account Type panel with regular Viewer user", () => {
  const publicUser = {
    id: 1,
    active: true,
    firstName: "Public",
    lastName: "User",
    email: "public.user@email.com",
    phone: "9995559999",
    role: 2
  }
  const tree = renderer.create(<ProfileType userData={publicUser} />)
  expect(tree).toMatchSnapshot()
})

it("renders Profile: Account Type panel with Passport user", () => {
  const passportUser = {
    id: 2,
    active: true,
    firstName: "Passport",
    lastName: "User",
    email: "passport.user@email.com",
    phone: "9995559999",
    role: 1
  }
  const tree = renderer.create(<ProfileType userData={passportUser} />)
  expect(tree).toMatchSnapshot()
})