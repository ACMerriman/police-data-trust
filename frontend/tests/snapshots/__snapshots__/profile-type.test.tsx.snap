// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Profile: Account Type panel with Passport user 1`] = `
<div>
  <div
    class="profileData"
  >
    <header
      class="sectionTitle"
    >
      Your Profile Type
    </header>
    <div
      class="profileTypeText"
    >
      Passport Account
    </div>
    <p
      class="description"
    >
      A Passport account allows you to see legally protected data. If the publicly available data is sufficient for your needs...
    </p>
    <div
      class="profileTypeLink"
    >
      <a
        href="/register"
      >
        Revert to a Viewer Account
      </a>
    </div>
  </div>
</div>
`;

exports[`renders Profile: Account Type panel with current user 1`] = `<div />`;
